File: shuttle-api/src/ShuttleApi.Domain/Trips/TripCargoItem.cs
using ShuttleApi.Domain.Common;

namespace ShuttleApi.Domain.Trips;

public sealed class TripCargoItem : Entity<Guid>
{
    public Guid TripId { get; private set; }
    public CargoType CargoType { get; private set; }
    public string? Description { get; private set; }
    public int Quantity { get; private set; }
    public decimal? WeightKg { get; private set; }
    public decimal? Charge { get; private set; }
    public bool IsHazmat { get; private set; }
    public bool IsSecured { get; private set; }

    private TripCargoItem() { }

    public static TripCargoItem Create(
        Guid tripId,
        CargoType cargoType,
        string? description,
        int quantity,
        decimal? weightKg = null,
        decimal? charge = null,
        bool isHazmat = false,
        bool isSecured = false)
    {
        Guard.Against(quantity < 1, "Quantity must be at least 1.");
        Guard.Against(weightKg.HasValue && weightKg.Value <= 0, "Weight must be greater than zero.");

        return new TripCargoItem
        {
            Id = Guid.NewGuid(),
            TripId = tripId,
            CargoType = cargoType,
            Description = description,
            Quantity = quantity,
            WeightKg = weightKg,
            Charge = charge,
            IsHazmat = isHazmat,
            IsSecured = isSecured
        };
    }
}

File: shuttle_tablet/lib/features/trips/domain/entities/trip_cargo_item.dart
import 'package:equatable/equatable.dart';

enum TripCargoType { box, pallet }

class TripCargoItem extends Equatable {
  final String id;
  final String tripId;
  final TripCargoType cargoType;
  final String? description;
  final int quantity;
  final double? weightKg;
  final double? charge;
  final bool isHazmat;
  final bool isSecured;

  const TripCargoItem({
    required this.id,
    required this.tripId,
    required this.cargoType,
    this.description,
    required this.quantity,
    this.weightKg,
    this.charge,
    this.isHazmat = false,
    this.isSecured = false,
  });

  String get typeLabel => switch (cargoType) {
        TripCargoType.box => 'Box',
        TripCargoType.pallet => 'Pallet',
      };

  @override
  List<Object?> get props =>
      [id, tripId, cargoType, description, quantity, weightKg, charge, isHazmat, isSecured];
}
